src/client/client.ts

```diff
@@ -50,6 +51,15 @@ export interface ClientOptions {
     /** Called to create the connection to the server. */
     createMessageTransports: () => MessageTransports | Promise<MessageTransports>
 
+    /**
+     * Called when the server requests to list files or get file contents. See
```

Comment:
Support for these should live entirely in a client feature. If we add things like this to the client itself, it will get unwieldy very fast.

Reply:
Gotcha, I'm going to follow the pattern of Observables of callbacks like `showInputRequest`.